### PR TITLE
Team 2 weapon into Main

### PR DESCRIPTION
**Pull request by Magnius001**
Merging `team-2-weapon` into `main`, added: 
 - Weapon rendering
 - Weapon animation
 - Melee weapon damage
 - Ammo counter
 - Weapon sound effects
 - Changes to `Collectible` and `ItemPickupComponent` to support passing `Entity` as a parameter
 - Remove `ConcreteRangedWeapon` and `ConcreteMeleeWeapon`

### CODE REVIEW

File: hs_err_pid77168.log
Comment (IanP96) [inline hs_err_pid77168.log:1]: should delete this I think, unless you need it

Comment (Magnius001) [reply hs_err_pid77168.log:1]: Ah right, I thought it other team's so didn't bother

Comment (Magnius001) [reply hs_err_pid77168.log:1]: Deleted.

**Comment by vsinh14** [comment]
LGTM

**Review by keagan-stoessel — COMMENTED** [review]
I approve

**Pull request merged**